api/src/main/java/org/pmiops/workbench/institution/InstitutionService.java

```diff
@@ -52,6 +52,25 @@ boolean validateAffiliation(
    */
   boolean validateInstitutionalEmail(Institution institution, String contactEmail);
 
+  /**
+   * Retrieve an ordered list of the email domains which this institution uses to match user contact
+   * email, or an empty list if none. Throws NotFoundException if the Institution does not exist.
+   *
+   * @param institutionShortName the short name (key) used to refer to this institution in the API
+   * @return the list of email domains associated with this institution, if any
+   */
+  List<String> getEmailDomains(String institutionShortName);
```

Comment:
So my thing is, why not just fetch an institution and then ask it for its email domains and addresses? I would expect the institution object to know (or expose how to get) this, but seeing it also on the service makes me wonder what the source of truth is.

Reply:
The mapper itself needs this in order to create the Institution